test/built-ins/Realm/prototype/evaluate/wrapped-function-from-return-values-share-no-identity.js

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2021 Rick Waldron. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-realm.prototype.evaluate
+description: >
+  Realm.prototype.evaluate wrapped function from return values share no identity.
+features: [callable-boundary-realms]
+---*/
+
+assert.sameValue(
+  typeof Realm.prototype.evaluate,
+  'function',
+  'This test must fail if Realm.prototype.evaluate is not a function'
+);
+
+const r = new Realm();
+
+r.evaluate(`
+function fn() { return 42; }
+globalThis.arrow = x => x * 2;
+globalThis.pFn = new Proxy(fn, {
+    apply() {
+        pFn.used = 1;
+        return 39;
+    }
+});
+async function aFn() {
+    return 1;
+}
+
+function * genFn() {
+    return 1;
+}
+
+fn.x = 'secrets';
+arrow.x = 'secrets';
+pFn.x = 'secrets';
+aFn.x = 'secrets';
+genFn.x = 'secrets';
+`)
+
+const wrappedOrdinary = r.evaluate('() => fn')();
+assert.sameValue(typeof wrappedOrdinary, 'function', 'ordinary function wrapped');
+assert.sameValue(wrappedOrdinary(), 42, 'ordinary, return');
+assert.sameValue(wrappedOrdinary.x, undefined, 'ordinary, no property shared');
```

Comment:
please add assertions verifying `x in wrapped*` and/or `Object.prototype.hasOwnProperty.call(wrappedOrdinary, 'x')`.

Reply:
I've added tests that check `Object.prototype.hasOwnProperty.call(wrappedOrdinary, 'x') === false`